xclim/core/missing.py

```diff
@@ -299,41 +275,112 @@ def missing_pct(da, freq, tolerance, **indexer):
     out : DataArray
       A boolean array set to True if period has missing values.
     """
-    return MissingPct(da, freq, **indexer)(tolerance=tolerance)
 
+    def is_missing(self, null, count, tolerance=0.1):
+        if tolerance < 0 or tolerance > 1:
+            raise ValueError("tolerance should be between 0 and 1.")
+
+        n = count - null.count(dim="time") + null.sum(dim="time")
+        return n / count >= tolerance
 
-def at_least_n_valid(da, freq, n=1, **indexer):
+    @staticmethod
+    def validate(tolerance):
+        return 0 <= tolerance <= 1
+
+
+@register_missing_method("at_least_n")
+class AtLeastNValid(MissingBase):
     r"""Return whether there are at least a given number of valid values.
 
-        Parameters
-        ----------
-        da : DataArray
-          Input array at daily frequency.
-        freq : str
-          Resampling frequency.
-        n : int
-          Minimum of valid values required.
-        **indexer : {dim: indexer, }, optional
-          Time attribute and values over which to subset the array. For example, use season='DJF' to select winter
-          values, month=1 to select January, or month=[6,7,8] to select summer months. If not indexer is given,
-          all values are considered.
+    Parameters
+    ----------
+    da : DataArray
+      Input array at daily frequency.
+    freq : str
+      Resampling frequency.
+    n : int
+      Minimum of valid values required.
+    **indexer : {dim: indexer, }, optional
+      Time attribute and values over which to subset the array. For example, use season='DJF' to select winter
+      values, month=1 to select January, or month=[6,7,8] to select summer months. If not indexer is given,
+      all values are considered.
 
-        Returns
-        -------
-        out : DataArray
-          A boolean array set to True if period has missing values.
-        """
-    return AtLeastNValid(da, freq, **indexer)(n=n)
+    Returns
+    -------
+    out : DataArray
+      A boolean array set to True if period has missing values.
+    """
 
+    def is_missing(self, null, count, n=20):
+        """The result of a reduction operation is considered missing if less than `n` values are valid."""
+        nvalid = null.count(dim="time") - null.sum(dim="time")
+        return nvalid < n
 
-def missing_from_context(da, freq, **indexer):
+    @staticmethod
+    def validate(n):
+        return n > 0
+
+
+@register_missing_method("skip")
+class Skip(MissingBase):
+    def __init__(self, da, freq=None, **indexer):
+        pass
+
+    def is_missing(self, null, count):
+        """Return whether or not the values within each period should be considered missing or not."""
+        return False
+
+    def __call__(self):
+        return False
+
+
+@register_missing_method("from_context")
```

Comment:
I'm not sure I understand how registering the "from_context" option works? Right now, this allows to do: `xc.set_options(check_missing='from_context')`, which will fail with an `NotImplementedError` as this class doesn't implement anything?

Reply:
You would tell an indicator to use the "from_context" missing method. Then, when calculating the mask, it would use the missing method from context. In your example, you're setting the context to "from_context", which is an infinite loop.

Reply:
Oh! Sorry I misread the code in #435! Then I suggest we modify the "check_missing" validator in `xclim.core.options` to forbid setting `from_context`, just in case.